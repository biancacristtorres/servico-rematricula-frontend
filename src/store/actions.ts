import { ActionTree } from 'vuex';
import { RootState } from '.';
import { AlunoMutationTypes } from './aluno/mutations';
import { StoreNamespaces } from './namespaces';
import { ResumoMutationTypes } from './resumo/mutations';
import { GlobalMutationTypes } from './mutations';
import { GlobalState } from './state';

export enum GlobalActionTypes {
  LIMPAR_STORE = 'LIMPAR_STORE',
  DEFINIR_LINK_PARA_VOLTAR = 'DEFINIR_LINK_PARA_VOLTAR',
  LIMPAR_LINK_PARA_VOLTAR = 'LIMPAR_LINK_PARA_VOLTAR',
}

const actions: ActionTree<GlobalState, RootState> = {

  async [GlobalActionTypes.LIMPAR_STORE]({ commit }) {
    commit(`${ StoreNamespaces.ALUNO }/${ AlunoMutationTypes.LIMPAR_STORE }`);
    commit(`${ StoreNamespaces.RESUMO }/${ ResumoMutationTypes.LIMPAR_STORE }`);
  },

  async [GlobalActionTypes.DEFINIR_LINK_PARA_VOLTAR]({ commit }, link: string) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, link);
  },

  async [GlobalActionTypes.LIMPAR_LINK_PARA_VOLTAR]({ commit }) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, '');
  },
};

export default actions;
