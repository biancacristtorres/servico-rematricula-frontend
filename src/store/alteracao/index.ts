import { Module } from 'vuex';
import { AlteracaoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const alteracao: Module<AlteracaoState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default alteracao;
