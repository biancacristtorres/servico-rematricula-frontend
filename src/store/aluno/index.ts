import { Module } from 'vuex';
import { AlunoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';


const Aluno: Module<AlunoState, RootState> = {
  actions,
  state,
  mutations,
  getters,
  namespaced: true,
};

export default Aluno;
