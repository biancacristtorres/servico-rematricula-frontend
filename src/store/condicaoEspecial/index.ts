import { Module } from 'vuex';
import { CondicaoEspecialState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const condicaoEspecial: Module<CondicaoEspecialState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default condicaoEspecial;
