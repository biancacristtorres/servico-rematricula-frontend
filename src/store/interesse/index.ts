import { Module } from 'vuex';
import { InteresseState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';


const interesse: Module<InteresseState, RootState> = {
  actions,
  state,
  mutations,
  getters,
  namespaced: true,
};

export default interesse;
