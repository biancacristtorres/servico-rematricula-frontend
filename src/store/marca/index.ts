import { Module } from 'vuex';
import { state, MarcaState } from './state';
import { RootState } from '@/store';
import getters from './getters';

const marca: Module<MarcaState, RootState> = {
  state,
  getters,
  namespaced: true,
};

export default marca;
