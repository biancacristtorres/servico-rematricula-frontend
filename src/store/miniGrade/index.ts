import { Module } from 'vuex';
import { state, MiniGradeState } from './state';
import { RootState } from '@/store';
import actions from './actions';
import mutations from './mutation';
import getters from './getters';

const miniGrade: Module<MiniGradeState, RootState> = {
  state,
  actions,
  mutations,
  getters,
  namespaced: true,
};

export default miniGrade;
