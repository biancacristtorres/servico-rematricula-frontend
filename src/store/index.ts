import Vue from 'vue';
import Vuex from 'vuex';
import autorizacao from 'componente-frontend-core/store/autorizacao';
import interesse from './interesse';
import condicaoEspecial from './condicaoEspecial';
import featureToggle from 'componente-frontend-core/store/featureToggle';
import alteracao from './alteracao';
import resumo from './resumo';
import aluno from './aluno';
import actions from './actions';
import marca from './marca';
import miniGrade from './miniGrade';
import { state } from './state';
import mutations from './mutations';

export { StoreNamespaces } from '@/store/namespaces';

Vue.use(Vuex);

export class RootState {
}

const modules = {
  autorizacao,
  interesse,
  condicaoEspecial,
  featureToggle,
  alteracao,
  resumo,
  aluno,
  marca,
  miniGrade,
};

const store = new Vuex.Store({
  modules,
  actions,
  state,
  mutations,
});

Object.assign(window, { $store: store });

export default store;
