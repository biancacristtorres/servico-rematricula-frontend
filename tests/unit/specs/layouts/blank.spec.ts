import { shallowMount } from '@vue/test-utils';
import blank from '@/layouts/blank.vue';
import store from '@/store';

describe('layout > black.vue', () => {
  it('deveRenderizar', () => {
    const wrapper = shallowMount(blank, {
      localVue,
      store,
      stubs: ['router-view'],
      mocks: {
        $route: {
          fullPath: '',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
