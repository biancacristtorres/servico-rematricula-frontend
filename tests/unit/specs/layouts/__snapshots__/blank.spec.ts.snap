// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout > black.vue deveRenderizar 1`] = `<router-view-stub></router-view-stub>`;
